src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -194,14 +206,141 @@ internal void UpdateSelectedItem(PackageItemListViewModel selectedItem)
+                    // So the retry button and message is displayed
+                    // Do not log to the activity log, since it is not a NuGet error
+                    _logger.Log(ProjectManagement.MessageLevel.Error, Resx.Resources.Text_UserCanceled);
+
+                        _loadingStatusIndicator.SetError(Resx.Resources.Text_UserCanceled);
+
+
+                        _loadingStatusBar.SetCancelled();
+                        _loadingStatusBar.Visibility = Visibility.Visible;
+                    }
+                    catch (Exception ex) when (!loadCts.IsCancellationRequested)
+                    {
+                        loadCts.Cancel();
+                        loadCts.Dispose();
+                        currentLoader.Reset();
+
+                    // Write stack to activity log
+                    Mvs.ActivityLog.LogError(LogEntrySource, ex.ToString());
+
+                        await _joinableTaskFactory.Value.SwitchToMainThreadAsync();
+
+                        var errorMessage = ExceptionUtilities.DisplayMessage(ex);
+                        _logger.Log(ProjectManagement.MessageLevel.Error, errorMessage);
+
+                        _loadingStatusIndicator.SetError(errorMessage);
+
+                        _loadingStatusBar.SetError();
+                        _loadingStatusBar.Visibility = Visibility.Visible;
+                    }
+                    catch (Exception ex)
+                    {
+                        var message = ex.ToString();
+                        LoadItemsCanceled?.Invoke(this, EventArgs.Empty);
+                        //throw;
+                    }
+                    finally
+                    {
+                        if (_loadingStatusIndicator.Status != LoadingStatus.NoItemsFound
+                            && _loadingStatusIndicator.Status != LoadingStatus.ErrorOccurred)
+                        {
+                        // Ideally, After a serach, it should report its status and,
+                        // do not keep the LoadingStatus.Loading forever.
+                        // This is a workaround.
+                        var emptyListCount = addedLoadingIndicator ? 1 : 0;
+
+                        //TODO: I think this can stay as-is because if there's a filter, the Items load still should finish.
+                        if (Items.Count == emptyListCount)
+                            {
+                                _loadingStatusIndicator.Status = LoadingStatus.NoItemsFound;
+                            }
+                            else
+                            {
+                                Items.Remove(_loadingStatusIndicator);
+                            }
+                        }
+
+                        UpdateCheckBoxStatus();
+                    }
+
+
+
+                    LoadItemsCompleted?.Invoke(this, EventArgs.Empty);
+                });
+                var exc = jtf.Task.Exception;
```

Comment:
First thing that stands out is that this won't work. 

The task is not awaited so this method will likely complete before the task above finished.

Reply:
Just hacks/debugging junk but good to know